### PR TITLE
[MARCHETYPES-81] generate target 8 instead of 1.8

### PR DESCRIPTION
**Pull request by hboutemy**
Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MARCHETYPES-81) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MARCHETYPES-XXX] SUMMARY`, where you replace `MARCHETYPES-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: maven-archetype-quickstart/src/test/resources/projects/it-java-7/archetype.properties [hunk 1]
```diff
@@ -2,4 +2,4 @@ groupId=archetype.it
 artifactId=java-8-quickstart
 version=0.1-SNAPSHOT
 package=it.pkg
-javaCompilerVersion=1.7
+javaCompilerVersion=7
```
Comment (judovana) [inline maven-archetype-quickstart/src/test/resources/projects/it-java-7/archetype.properties:5]: are you sure "7" will be recognized?

File: maven-archetype-quickstart/src/test/resources/projects/it-java-7/reference/pom.xml [hunk 2]
```diff
@@ -13,8 +13,8 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.7</maven.compiler.source>
-    <maven.compiler.target>1.7</maven.compiler.target>
+    <maven.compiler.source>7</maven.compiler.source>
+    <maven.compiler.target>7</maven.compiler.target>
```
Comment (judovana) [inline maven-archetype-quickstart/src/test/resources/projects/it-java-7/reference/pom.xml:17]: likewise

File: maven-archetype-quickstart/src/test/resources/projects/it-java-7/archetype.properties (hunk 1 above)
Comment (hboutemy) [reply maven-archetype-quickstart/src/test/resources/projects/it-java-7/archetype.properties:5]: you can build and see it works
but this is a good question that I had myself previously: what precisely makes this work? I summarised my findings in https://issues.apache.org/jira/browse/MPOM-449

Comment (judovana) [reply maven-archetype-quickstart/src/test/resources/projects/it-java-7/archetype.properties:5]: Thanx!

**Comment by jira-importer** [comment]
Resolve #114

**Pull request merged**